gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/DefaultKeystoreService.java

```diff
@@ -333,6 +336,7 @@ public void removeCredentialForCluster(String clusterName, String alias) throws
 
         final Path keyStoreFilePath = keyStoreDirPath.resolve(clusterName + CREDENTIALS_SUFFIX);
         writeKeyStoreToFile(ks, keyStoreFilePath, masterService.getMasterSecret());
+        removeFromCache(clusterName, alias);
```

Comment:
We want this to happen no matter what right? Maybe in the finally block or leave it at the beginning?

Reply:
Why? Should not we only remove it from cache only when it was removed from the keystore successfully?